apps/ledger-live-desktop/src/newArch/features/WalletSync/hooks/useQRCode.ts

```diff
@@ -26,73 +33,76 @@ export function useQRCode() {
+    mutationFn: (memberCredentials: MemberCredentials) =>
+      createQRCodeHostInstance({
+        trustchainApiBaseUrl,
+        onDisplayQRCode: url => {
+          setUrl(url);
+        },
+        onDisplayDigits: digits => {
+          dispatch(setQrCodePinCode(digits));
+          dispatch(setFlow({ flow: Flow.Synchronize, step: Step.PinCode }));
+        },
+        addMember: async member => {
+          if (trustchain) {
+            await sdk.addMember(trustchain, memberCredentials, member);
+            return trustchain;
+          }
+          throw new NoTrustchainInitialized();
+        },
+        memberCredentials,
+        memberName,
+        alreadyHasATrustchain: !!trustchain,
+      }),
 
-    setError(null);
-    setIsLoading(true);
-    createQRCodeHostInstance({
-      trustchainApiBaseUrl,
-      onDisplayQRCode: url => {
-        setUrl(url);
-      },
-      onDisplayDigits: digits => {
-        dispatch(setQrCodePinCode(digits));
-        dispatch(setFlow({ flow: Flow.Synchronize, step: Step.PinCode }));
-      },
-      addMember: async member => {
-        if (trustchain) {
-          await sdk.addMember(trustchain, memberCredentials, member);
-          return trustchain;
-        }
-        throw new NoTrustchainInitialized();
-      },
-      memberCredentials,
-      memberName,
-      alreadyHasATrustchain: !!trustchain,
-    })
-      .catch(e => {
-        if (e instanceof InvalidDigitsError) {
-          dispatch(setFlow({ flow: Flow.Synchronize, step: Step.PinCodeError }));
-        } else if (e instanceof NoTrustchainInitialized) {
-          dispatch(setFlow({ flow: Flow.Synchronize, step: Step.UnbackedError }));
-        }
-        setError(e);
-        throw e;
-      })
-      .then(newTrustchain => {
-        if (newTrustchain) {
-          dispatch(setTrustchain(newTrustchain));
-        }
-        dispatch(
-          setFlow({
-            flow: Flow.Synchronize,
-            step: Step.SynchronizeLoading,
-            nextStep: Step.Synchronized,
-            hasTrustchainBeenCreated: false,
-          }),
-        );
-        queryClient.invalidateQueries({ queryKey: [QueryKey.getMembers] });
-        setUrl(null);
-        dispatch(setQrCodePinCode(null));
-        setIsLoading(false);
-        setError(null);
-      });
-  }, [memberCredentials, trustchainApiBaseUrl, memberName, dispatch, trustchain, sdk, queryClient]);
+    // Don't use retry here because it always uses a delay despite setting it to 0
+    onError(e) {
+      if (e instanceof QRCodeWSClosed) {
+        const { time } = e as unknown as { time: number };
+        if (time >= MIN_TIME_TO_REFRESH) startQRCodeProcessing();
+      }
+      if (e instanceof InvalidDigitsError) {
+        dispatch(setFlow({ flow: Flow.Synchronize, step: Step.PinCodeError }));
+      }
+      if (e instanceof NoTrustchainInitialized) {
+        dispatch(setFlow({ flow: Flow.Synchronize, step: Step.UnbackedError }));
+      }
```

Comment:
we need to add 

`  setError(e);
        throw e;` also no? Like before

Reply:
The `setError` is not needed because react-query manages the error (that's why I removed the `useState`).

For the `throw` I don't think it's needed either. From my understanding it was here to avoid the `then` that follows. Or am I missing something ?